core/src/main/java/com/graphhopper/routing/util/PrepareRoutingSubnetworks.java

```diff
@@ -73,45 +68,52 @@ public PrepareRoutingSubnetworks setMinOneWayNetworkSize( int minOnewayNetworkSi
 
     public void doWork()
     {
-        int del = removeZeroDegreeNodes();
+        if (minNetworkSize <= 0 && minOneWayNetworkSize <= 0)
+            return;
 
-        int unvisitedDeadEnds = -1;
-        if (minOneWayNetworkSize > 0 && singleEncoder != null)
+        int unvisitedDeadEnds = 0;
+        for (FlagEncoder encoder : encoders)
         {
-            unvisitedDeadEnds = removeDeadEndUnvisitedNetworks(singleEncoder);
+            // mark edges for one vehicle as inaccessible
+            PrepEdgeFilter filter = new PrepEdgeFilter(encoder);
+            if (minOneWayNetworkSize > 0)
+                unvisitedDeadEnds += removeDeadEndUnvisitedNetworks(filter);
+
+            List<TIntArrayList> components = findSubnetworks(filter);
+            keepLargeNetworks(filter, components);
+            subnetworks = Math.max(components.size(), subnetworks);
```

Comment:
The idea is to make the subnetwork calculation separate for every vehicle via the access restrictions instead of node removal.

Still the node removal occurs later on (in markNodesRemovedIfUnreachable) if no vehicle has access to the adjacent edges.